poly-commitment/src/tests/serialization.rs

```diff
@@ -0,0 +1,160 @@
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    ];
+    test_generic_serialization_regression_serde(data_expected, buf_expected);
+}
+
+#[test]
+pub fn ser_regression_canonical_polycomm() {
+    use crate::{commitment::BlindedCommitment, srs::SRS};
+    use ark_poly::{univariate::DensePolynomial, DenseUVPolynomial};
+    use mina_curves::pasta::{Fp, Vesta};
+
+    let rng = &mut StdRng::from_seed([0u8; 32]);
+
+    let srs = SRS::<Vesta>::create(1 << 7);
+
+    let com_length = 300;
+    let num_chunks = 6;
+
+    let poly = DensePolynomial::<Fp>::rand(com_length, rng);
+
+    let BlindedCommitment {
+        commitment: chunked_commitment,
+        blinders: _blinders,
+    } = srs.commit(&poly, num_chunks, rng);
+
+    let data_expected = chunked_commitment;
+    // Generated with commit 1494cf973d40fb276465929eb7db1952c5de7bdc
+    let buf_expected: Vec<u8> = vec![
+        145, 150, 196, 33, 36, 158, 70, 161, 147, 233, 138, 19, 54, 52, 87, 58, 158, 154, 255, 197,
+        219, 225, 79, 25, 41, 193, 232, 64, 250, 71, 230, 154, 34, 145, 81, 23, 0, 196, 33, 37,
+        227, 246, 88, 42, 44, 53, 244, 102, 92, 197, 246, 56, 56, 135, 155, 248, 155, 243, 23, 76,
+        44, 94, 125, 60, 209, 195, 190, 73, 158, 97, 42, 128, 196, 33, 101, 254, 242, 100, 238,
+        214, 56, 151, 94, 170, 69, 219, 239, 135, 253, 151, 207, 217, 47, 229, 75, 7, 41, 9, 131,
+        205, 85, 171, 166, 213, 96, 52, 128, 196, 33, 53, 166, 32, 175, 166, 196, 121, 1, 25, 236,
+        34, 226, 31, 145, 70, 96, 89, 179, 65, 35, 253, 161, 10, 211, 170, 116, 247, 40, 225, 104,
+        155, 34, 0, 196, 33, 114, 151, 94, 73, 26, 234, 37, 98, 188, 142, 161, 165, 62, 238, 58,
+        76, 200, 16, 62, 210, 124, 127, 229, 81, 119, 145, 43, 157, 254, 237, 154, 57, 128, 196,
+        33, 212, 213, 38, 1, 17, 84, 147, 102, 31, 103, 242, 177, 110, 64, 239, 33, 211, 216, 40,
+        103, 51, 55, 85, 96, 133, 20, 194, 6, 87, 180, 212, 13, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0,
+    ];
+
+    test_generic_serialization_regression_serde(data_expected, buf_expected);
+}
+
+#[test]
+pub fn ser_regression_canonical_opening_proof() {
+    use crate::evaluation_proof::OpeningProof;
+    use groupmap::GroupMap;
+    use mina_curves::pasta::Vesta;
+
+    let rng = &mut StdRng::from_seed([0u8; 32]);
+
+    let group_map = <Vesta as CommitmentCurve>::Map::setup();
+    let srs = SRS::<Vesta>::create(1 << 7);
+
+    let data_expected: OpeningProof<Vesta> =
+        crate::tests::commitment::generate_random_opening_proof(rng, &group_map, &srs).0[0]
+            .proof
+            .clone();
+
+    // Generated with commit 1494cf973d40fb276465929eb7db1952c5de7bdc
+    let buf_expected: Vec<u8> = vec![
+        149, 151, 146, 196, 33, 166, 183, 76, 209, 121, 62, 56, 194, 134, 133, 238, 52, 190, 163,
```

Comment:
Nit: could be nice to load the value from a file

Reply:
I'd rather keep it here tbh. I know it's big and ugly, but file management is more annoying imo, you rely on extra kernel calls, it's more files, and if we change serialization in a certain way git would probably just think the whole binary file was replaced, even if it's just a bit. The last case we had before actually when I was replacing the SRS serialization, you could clearly see that the new serialization is just one bit off, which helps with debugging.